x/bep3/genesis_test.go

```diff
@@ -278,19 +278,14 @@ func (suite *GenesisTestSuite) TestGenesisState() {
 	}
 
 	for _, tc := range testCases {
-		suite.SetupTest()
 		if tc.expectPass {
-			suite.Run(tc.name, func() {
```

Comment:
We added `suite.Run(tc.name, func() ... )` because the test suite was reusing state and swallowing panics. Tested a few cases and the panics were caught due to genesis param validation, but want @karzak to take a look before approving.

Reply:
It also looks like tests are now failing because `ClosedBlock` can't be 0. For example, `name: "atomic swap asset type is unsupported"` actually fails due to `failed to validate bep3 genesis state: closed block cannot be 0`.

Can resolve by updating `ClosedBlock` to 1 for each test case, but I recommend checking the error message so we catch similar issues in the future. I can do it throughout the BEP3 module if you'd like as it's a legacy issue and adding the expected error msg to each test case is pretty tedious.

Reply:
👍 to checking the error message to catch test failing for reasons other than the intended reason. Can be in a follow up PR.

I had the same thoughts about `suite.Run, but indeed the tests are catching things properly here. Not sure if it was use of `Suite.Nil` versus `suite.Require()` methods that was causing the problem previously. Seems fine to merge as is though.

Reply:
yeah, that's why I changed it to require. Otherwise, it reuses the state. We can do a test cleanup in another PR since it's out of scope from this one

Reply:
Created issue https://github.com/Kava-Labs/kava/issues/498 for the test cleanup